src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -374,6 +374,13 @@ else if (qp.getType() == Leader.SNAP) {
                 // The leader is going to dump the database
                 // db is clear as part of deserializeSnapshot()
                 zk.getZKDatabase().deserializeSnapshot(leaderIs);
+                // ZOOKEEPER-2819: overwrite config node content extracted
+                // from leader snapshot with local config, to avoid potential
+                // inconsistency of config node content during rolling restart.
+                if (!QuorumPeerConfig.isReconfigEnabled()) {
```

Comment:
@shralex instead of avoiding deserializing the config zNode we could overwrite the deserialized zNode content with local quorum config. Test will be covered in the test improvement JIRA ZOOKEEPER-2828.